crates/core/src/types.rs

```diff
@@ -10,6 +10,69 @@ use uuid::Uuid;
 /// Block time in microseconds
 pub const BLOCK_TIME: u128 = 2_000_000;
 
+pub struct BundleTransactions(Vec<Bytes>);
+
+impl From<Vec<Bytes>> for BundleTransactions {
+    fn from(txs: Vec<Bytes>) -> Self {
+        BundleTransactions(txs)
+    }
+}
+
+impl BundleTransactions {
+    pub fn bundle_hash(&self) -> Result<B256, String> {
```

Comment:
need to think about the return type, not a big fan of return Results for everything due to `self.transactions()`